include/entropy.h

```diff
@@ -64,6 +75,25 @@ static inline int _impl_entropy_get_entropy(struct device *dev,
 	return api->get_entropy(dev, buffer, length);
 }
 
+/**
+ * @brief Fills a buffer with currently available entropy.
+ *        Callable from Interrupt Service Routines.
+ *
+ * @param dev Pointer to the entropy device.
+ * @param buffer Buffer to fill with entropy.
+ * @param length Buffer length.
+ * @retval bytes filled on success.
+ * @retval -ERRNO errno code on error.
+ */
+static inline int entropy_isr_get_entropy(struct device *dev, u8_t *buffer,
+					  u16_t length)
+{
+	const struct entropy_driver_api *api = dev->driver_api;
+
+	__ASSERT(api->isr_get_entropy, "Callback pointer should not be NULL");
```

Comment:
I wonder if we should do something like:
```
if (api->isr_get_entropy) {
   return ...
} else {
   return -ENOTSUP;
}
```
Since not every driver might support this.

Reply:
Agreed, will change

Reply:
this is one additional branching in ISR.

Reply:
Only with kernel asserts enabled (`CONFIG_DEBUG`)

Reply:
@cvinayak Sorry, I misread. @galak on second thought this is an ISR call, so usually critical. I'd rather __ASSERT() than return an error here.